src/test/regress/sql/multi_partitioning_utils.sql

```diff
@@ -0,0 +1,218 @@
+ALTER SEQUENCE pg_catalog.pg_dist_shardid_seq RESTART 1620000;
+ALTER SEQUENCE pg_catalog.pg_dist_jobid_seq RESTART 1620000;
+
+
+-- ===================================================================
+-- create test functions
+-- ===================================================================
+
+CREATE FUNCTION generate_alter_table_attach_partition_command(regclass)
+    RETURNS text
+    AS 'citus'
+    LANGUAGE C STRICT;
+
+CREATE FUNCTION generate_partition_information(regclass)
+    RETURNS text
+    AS 'citus'
+    LANGUAGE C STRICT;
+
+CREATE FUNCTION print_partitions(regclass)
+    RETURNS text
+    AS 'citus'
+    LANGUAGE C STRICT;
+
+CREATE OR REPLACE FUNCTION detach_and_attach_partition(partition_name regclass, parent_table_name regclass)
+RETURNS void LANGUAGE plpgsql VOLATILE
+AS $function$
+DECLARE
+	attach_partition_command text := '';
+	command_result text := '';
+
+BEGIN
+   -- first generate the command
+   SELECT public.generate_alter_table_attach_partition_command(partition_name) INTO attach_partition_command;
+
+   -- later detach the same partition
+   EXECUTE 'ALTER TABLE ' || parent_table_name || ' DETACH PARTITION ' || partition_name || ';';
+
+   -- not attach it again
+   EXECUTE attach_partition_command;
+END;
+$function$;
+
+CREATE OR REPLACE FUNCTION drop_and_recreate_partitioned_table(parent_table_name regclass)
+RETURNS void LANGUAGE plpgsql VOLATILE
+AS $function$
+DECLARE
+	command text := '';
+
+BEGIN
+   -- first generate the command
+   CREATE TABLE partitioned_table_create_commands AS SELECT master_get_table_ddl_events(parent_table_name::text);
+
+   -- later detach the same partition
+   EXECUTE 'DROP TABLE ' || parent_table_name::text || ';';
+
+   FOR command IN SELECT * FROM partitioned_table_create_commands
```

Comment:
Ah, makes sense. Probably makes more sense to use an array, but this is also fine.